sky/backends/monkey_patches/monkey_patch_ray_up.py

```diff
@@ -14,11 +14,14 @@
 
 Reasons:
  - With our patch, the ray up will always reuse the existing head node with the same cluster
- name, even if the launch hash is different to avoid resources leakage. The outer-level
- SkyPilot code will guarantee that reusing the existing head node is safe, as the ray yaml
- is generated by SkyPilot (not modified by the user), i.e. the case where the launch hash
- is used to detect a change in the node_config is not needed.
-
+   name, even if the launch hash is different to avoid resources leakage. The outer-level
+   SkyPilot code will guarantee that reusing the existing head node is safe, as the ray yaml
+   is generated by SkyPilot (not modified by the user), i.e. the case where the launch hash
+   is used to detect a change in the node_config is not needed.
+   For worker nodes, ray autoscaler will check the launch hash by calling
+   ray.autoscaler._private.autoscaler.launch_config_ok(). We pass the
+   disable_launch_config_check to the generated ray yaml (sky.templates.<cloud>-ray.yml.j2)
+   to avoid the launch hash check for worker nodes.
```

Comment:
Also, when restarting a multinode cluster, the logic for considering what worker nodes to restart/"reuse" is in node_provider.py's create_node(), the same code for deciding what head node(s) to restart/"reuse".